src/gtc/gtcpp/gtcpp.py

```diff
@@ -246,12 +241,6 @@ class GTStage(LocNode):
     # or `temporaries`
     args: List[Arg]
 
-    @validator("args")
```

Comment:
Why did you remove it? I think if this is removed we should add a test to `test_gtcpp_compilation.py` that ensures a stage with no arguments produces compilable code.

Reply:
Because there is _test_stage_without_effect_ that assigns to a temporary and does nothing else. After the temporary is replaced by a scalar in the optimization passes, there are no arguments left. And it works, so this test basically exists.

Reply:
Can you still add an explicit test.

Reply:
Done.